### ISSUE TITLE
F841 not reporting unused var in for loops

### ISSUE THREAD

**Issue by sk-** [astral-sh/ruff#16754]
### Summary

F841 is not reporting unused loop vars as unused. This can be seen in the following code ([playground](https://play.ruff.rs/f5ecf49d-df91-4033-adaf-a7ad42c39f07))

```python
def func():
    for x in []:
        pass

    data = {}
    for key1, val1 in data.items():
        pass

    for key2, val2 in data.items():
        print(key2)

    for ke3, val3 in data.items():
        print(val3)
```
In the code above, if the variables were reported as unused, the programmer, could be able to realize that better constructs exist for their use case, replacing the last two loops with

```python
     for key2 in data:
        print(key2)

    for val3 in data.values():
        print(val3)
```

May be related to #16753. 

### Version

v0.11.0 (playground)
